### PR TITLE
Add support for darwin/clang

### PR DESCRIPTION
**Pull request by kiwiroy**
This pull request adds support for clang on macOS. This involves a switch to a more `Alien::Build` style, including an `alienfile`, compiling at `probe` time with `Probe::CBuilder`, and only supporting system installs. AB manages the runtime settings, making `Alien::OpenMP` more simple. The other side of this is that there's a new module to store the possible scenarios and a dependency on `Alien::Build`.

Thanks for considering this request.

### CODE REVIEW

File: lib/Alien/OpenMP.pm [hunk 1]
```diff
@@ -1,82 +1,26 @@
...
-our $omp_check_libs = {
-    # used by _check_libs, intended for use by Devel::CheckLib
-    gcc => [qw/gomp/],
-};
-
-our $omp_check_headers = {
-    # used by _check_headers, intended for use by Devel::CheckLib
-    gcc => [qw/omp.h/],
-};
+our $VERSION = '0.003003';
```
Comment (mohawk2) [inline lib/Alien/OpenMP.pm]: Could you undo this version change? That's for release person to do :-)

**Review by mohawk2 — CHANGES_REQUESTED** [review]
This looks really good overall! I've enabled the CI.

File: lib/Alien/OpenMP.pm (hunk 1 above)
Comment (kiwiroy) [reply lib/Alien/OpenMP.pm]: done

**Review by mohawk2 — APPROVED** [review]

**Comment by mohawk2** [comment]
Thanks for this! Released as 0.003003.

**Comment by oodler577** [comment]
thank you, @kiwiroy and @mohawk2

**Pull request merged**